src/client-side-encryption/state_machine.ts

```diff
@@ -289,14 +302,27 @@ export class StateMachine {
   async kmsRequest(request: MongoCryptKMSRequest): Promise<void> {
     const parsedUrl = request.endpoint.split(':');
     const port = parsedUrl[1] != null ? Number.parseInt(parsedUrl[1], 10) : HTTPS_PORT;
-    const options: tls.ConnectionOptions & { host: string; port: number } = {
+    const options: tls.ConnectionOptions & {
+      host: string;
+      port: number;
+      autoSelectFamily?: boolean;
+      autoSelectFamilyAttemptTimeout?: number;
+    } = {
       host: parsedUrl[0],
       servername: parsedUrl[0],
       port
     };
     const message = request.message;
     const buffer = new BufferPool();
 
+    const socketOptions = this.options.socketOptions || {};
+    if ('autoSelectFamily' in socketOptions) {
+      options.autoSelectFamily = socketOptions.autoSelectFamily;
+    }
+    if ('autoSelectFamilyAttemptTimeout' in socketOptions) {
+      options.autoSelectFamilyAttemptTimeout = socketOptions.autoSelectFamilyAttemptTimeout;
+    }
+
     const netSocket: net.Socket = new net.Socket();
```

Comment:
commenting here, but the code I'm thinking about is lowerdown. Should `netSocket.connect` also get this option? even though it is targeting the proxy I think you'd still want the consistent family selection behavior right? 

Does the same apply for SOCKS5 inside the `connect.ts` file? I think it does already because the socks5 negotiates over socket we will have made with the autoSelectFamily option set, but good to double check.

Reply:
1. I've updated fot `netSocket.connect` to get these options now.
2. Yes the raw socket for socks5 is going through `makeSocket` which will have the options provided.